cardano-explorer/cardano-explorer.cabal

```diff
@@ -21,14 +21,19 @@ library
   default-language:     Haskell2010
   hs-source-dirs:       src
   ghc-options:          -Wall
+                        -Werror
                         -Wcompat
                         -fwarn-redundant-constraints
                         -fwarn-incomplete-patterns
                         -fwarn-unused-imports
                         -Wincomplete-record-updates
                         -Wincomplete-uni-patterns
                         -Weverything
-                        -- -Wno-unused-matches
+                        -Wno-safe
+                        -Wno-unsafe
+                        -Wno-implicit-prelude
+                        -Wno-error=partial-fields
```

Comment:
We *always* want partial-fields to be both a warning and an error.

Reply:
Is there no workaround to avoid partial fields?

Reply:
If there is no alternative, then we use the `OPTIONS_GHC` pragma on the files that need it rather than enable it for the whole project.

Reply:
we could make it a sum type that contains 2 new data records, and then have custom `toJSON` instances to maintain the old encoding
i'm thinking just leave it as it is now (plus the `OPTIONS_GHC`), then look into it more when we implement that api call